core/src/main/java/com/radixdlt/api/system/routes/HealthHandler.java

```diff
@@ -66,22 +66,21 @@
 
 import com.google.inject.Inject;
 import com.radixdlt.api.system.SystemGetJsonHandler;
-import com.radixdlt.api.system.generated.models.HealthResponse;
-import com.radixdlt.api.system.generated.models.RecentSelfProposalMissStatistic;
+import com.radixdlt.api.system.generated.models.*;
 import com.radixdlt.api.system.health.HealthInfoService;
-import com.radixdlt.protocol.Current;
-import com.radixdlt.protocol.ProtocolVersion;
+import com.radixdlt.protocol.ProtocolUpdateEnactmentCondition;
+import com.radixdlt.statecomputer.ProtocolState;
+import com.radixdlt.statecomputer.RustStateComputer;
 
 public final class HealthHandler extends SystemGetJsonHandler<HealthResponse> {
   private final HealthInfoService healthInfoService;
-  private final ProtocolVersion currentProtocolVersion;
+  private final RustStateComputer rustStateComputer;
 
   @Inject
-  HealthHandler(
-      HealthInfoService healthInfoService, @Current ProtocolVersion currentProtocolVersion) {
+  HealthHandler(HealthInfoService healthInfoService, RustStateComputer rustStateComputer) {
```

Comment:
Any way we can remove this dependency on `RustStateComputer` for example by using messages? Seems pretty bad to have this API contend with the protocol state lock with the state computer.

Otherwise, this is probably not too bad of a hack but we should add a comment.

Reply:
Good point! Gonna take a look at that

Reply:
Done: I reused the LedgerUpdate event